src/features/slackapi/slackapi-utils.ts

```diff
@@ -123,20 +137,33 @@ Have a great week!
    * Post a slack message to users
    *
    * @param weeklyCombinedData list of combined timebank and slack user data
-   * @param weekStart when time data starts
-   * @param weekEnd when time data ends
+   * @param nonProjectTimes all non project times
+   * @param timeRegistrations all time registrations after yesterday
+   * @param previousWorkDays dates and the number of today
    */
-  export const postWeeklyMessage = (weeklyCombinedData: WeeklyCombinedData[], weekStart: string, weekEnd: string) => {
+  export const postWeeklyMessage = (
+    weeklyCombinedData: WeeklyCombinedData[],
+    timeRegistrations:TimeRegistrations[],
+    previousWorkDays: PreviousWorkdayDates,
+    nonProjectTimes: NonProjectTime[]) => {
+    const { weekStartDate, weekEndDate } = TimeUtilities.lastWeekDateProvider();
+    const { yesterday, today } = previousWorkDays;
+
     weeklyCombinedData.forEach(user => {
-      const { slackId } = user;
-
-      try {
-        client.chat.postMessage({
-          channel: slackId,
-          text: constructWeeklySummaryMessage(user, weekStart, weekEnd)
-        });
-      } catch (error) {
-        console.error(`Error while posting weekly slack messages to user ${user.name}`);
+      const { slackId, personId } = user;
+
+      const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, 435, today, nonProjectTimes);
+      const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, 435, yesterday, nonProjectTimes);
```

Comment:
Should probably use expected time for week instead of hard coding values

Reply:
Done